src/function/special/zeta.js

```diff
@@ -0,0 +1,140 @@
+import { factory } from '../../utils/factory.js'
+
+const name = 'zeta'
+const dependencies = ['typed', 'config', 'multiply', 'pow', 'divide', 'factorial', 'equal', 'gamma', 'sin', 'subtract', 'add', 'Complex', 'BigNumber', 'pi']
+
+export const createZeta = /* #__PURE__ */ factory(name, dependencies, ({ typed, config, multiply, pow, divide, factorial, equal, gamma, sin, subtract, add, Complex, BigNumber, pi }) => {
+  /**
+   * Compute the Riemann Zeta function of a value using an infinite series for
+   * all of the complex plane using Riemann's Functional equation.
+   *
+   * Based off the paper by Xavier Gourdon and Pascal Sebah
+   * ( http://numbers.computation.free.fr/Constants/Miscellaneous/zetaevaluations.pdf )
+   *
+   * Implementation and slight modification by Anik Patel
+   *
+   *
+   * Syntax:
+   *
+   *    math.zeta(n)
+   *
+   * Examples:
+   *
+   *    math.zeta(5)       // returns 1.0369277551433699...
+   *    math.zeta(-0.5)    // returns -0.2078862249773...
+   *    math.zeta(math.i)  // returns 0.0033002..., -0.4181554491...i
+   *
+   *
+   * @param {number | Complex | BigNumber} s   A Real, Complex or BigNumber parameter to to the Riemann Zeta Function
+   * @return {number | Complex | BigNumber}    The Riemann Zeta of `s`
+   */
+  function zeta (s) {
+    if (s.re === 0 && s.im === 0) {
+      return -0.5
+    }
+    if (s.re === 1) {
+      return NaN
+    }
+    if (s.re === Infinity && s.im === 0) {
+      return 1
+    }
+    if (s.im === Infinity || s.re === -Infinity) {
+      return NaN
+    }
+    // The "15" is number of decimal places of accuracy (approx)
+    const n = Math.round(1.3 * 15 + 0.9 * Math.abs(s.im))
+    if (s.re > -(n - 1) / 2) { return f(s, n) }
+
+    // Functional Equation for reflection to re(s) < -( n - 1) / 2
+    let c = multiply(pow(2, s), pow(Math.PI, subtract(s, 1)))
+    c = multiply(c, (sin(multiply(Math.PI / 2, s))))
+    c = multiply(c, gamma(subtract(1, s)))
+    return multiply(c, zeta(subtract(1, s)))
+  }
+  // Big Number alias
+  function zetaBigNumber (x) {
+    if (x === 0) {
+      return -0.5
+    }
+    if (equal(x, 1)) {
+      return NaN
+    }
+    if (!x.isFinite() && !x.isNegative()) {
+      return 1
+    }
+    if (!x.isFinite() && x.isNegative()) {
+      return NaN
+    }
+    // 15 decimals of accuracy
+    const n = 20
+    if (x > -(n - 1) / 2) { return fBigNumber(x, n) }
```

Comment:
There are still a few places where you implicitly convert a BigNumber into anumber, like `x === 0` and `x > -(n - 1) / 2`. I think they cannot hurt in practice since `n` is small and integer, but it would be neat to use `equal(x, 0)` and `larger(x, -(n - 1) / 2)` I think.